types/roslib/index.d.ts

```diff
@@ -77,173 +78,258 @@ declare namespace ROSLIB {
          */
-        getTopicType(topic: string, callback: (type: string) => void, failedCallback?: (error: any) => void): void;
+        getTopicType(topic: string, callback: (type: string) => void, failedCallback?: (error: string) => void): void;
 
         /**
-         * Retrieves a type of ROS service.
+         * Retrieve the type of a ROS service.
          *
-         * @param service name of service:
-         * @param callback - function with params:
-         *   * type - String of the service type
-         * @param failedCallback - the callback function when the ros call failed (optional). Params:
-         *   * error - the error message reported by ROS
+         * @param {string} service - Name of the service.
+         * @param {function} callback - Function with the following params:
+         * @param {string} callback.type - The type of the service.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
          */
-        getServiceType(service: string, callback: (type: string) => void, failedCallback?: (error: any) => void): void;
+        getServiceType(service: string, callback: (type: string) => void, failedCallback?: (error: string) => void): void;
 
         /**
-         * Retrieves a detail of ROS message.
+         * Retrieve the details of a ROS message.
          *
-         * @param callback - function with params:
-         *   * details - Array of the message detail
-         * @param message - String of a topic type
-         * @param failedCallback - the callback function when the ros call failed (optional). Params:
-         *   * error - the error message reported by ROS
+         * @param {string} message - The name of the message type.
+         * @param {function} callback - Function with the following params:
+         * @param {string} callback.details - An array of the message details.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
          */
         getMessageDetails(
-            message: Message,
-            callback: (detail: any) => void,
-            failedCallback?: (error: any) => void,
+            message: string,
+            callback: (details: string) => void,
+            failedCallback?: (error: string) => void,
         ): void;
 
         /**
-         * Decode a typedefs into a dictionary like `rosmsg show foo/bar`
+         * Decode a typedef array into a dictionary like `rosmsg show foo/bar`.
          *
-         * @param defs - array of type_def dictionary
+         * @param {any[]} defs - Array of type_def dictionary.
          */
-        decodeTypeDefs(defs: any): void;
+        decodeTypeDefs(defs: any[]): void;
```

Comment:
This method has a [return value](https://github.com/RobotWebTools/roslibjs/blob/develop/src/core/Ros.js#L688)